live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/TypeFilterApplierTest.java

```diff
@@ -0,0 +1,168 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.hxi_connector.live_ingester.adapters.messaging.repository.filter;
+
+import static java.util.Collections.emptyList;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.given;
+
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import org.alfresco.hxi_connector.live_ingester.adapters.config.properties.Filter;
+import org.alfresco.repo.event.v1.model.DataAttributes;
+import org.alfresco.repo.event.v1.model.NodeResource;
+import org.alfresco.repo.event.v1.model.RepoEvent;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class TypeFilterApplierTest
+{
+    @Mock
+    private RepoEvent<DataAttributes<NodeResource>> mockRepoEvent;
+    @Mock
+    private DataAttributes<NodeResource> mockData;
+    @Mock
+    private NodeResource mockResource;
+    @Mock
+    private Filter mockFilter;
+    @Mock
+    private Filter.Type mockType;
+
+    @InjectMocks
+    private TypeFilterApplier objectUnderTest;
+
+    @BeforeEach
+    void mockBasicData()
+    {
+        given(mockRepoEvent.getData()).willReturn(mockData);
+        given(mockData.getResource()).willReturn(mockResource);
+        given(mockFilter.type()).willReturn(mockType);
+    }
+
+    @Test
+    void shouldNotFilterOutWhenEmptyAllowedAndEmptyDenied()
+    {
+        given(mockResource.getNodeType()).willReturn("cm:folder");
+        given(mockType.allow()).willReturn(emptyList());
+        given(mockType.deny()).willReturn(emptyList());
+
+        // when
+        boolean result = objectUnderTest.applyFilter(mockRepoEvent, mockFilter);
+
+        // then
+        assertTrue(result);
+    }
+
+    @Test
+    void shouldNotFilterOutWhenTypeInAllowedAndEmptyDenied()
+    {
+        final String type1 = "cm:content";
+        final String type2 = "cm:special-folder";
+        given(mockResource.getNodeType()).willReturn(type1);
+        given(mockType.allow()).willReturn(List.of(type1, type2));
+        given(mockType.deny()).willReturn(emptyList());
+
+        // when
+        boolean result = objectUnderTest.applyFilter(mockRepoEvent, mockFilter);
+
+        // then
+        assertTrue(result);
+    }
+
+    @Test
+    void shouldNotFilterOutWhenEmptyAllowedAndTypeNotInDenied()
+    {
+        final String type1 = "cm:folder";
+        final String type2 = "cm:special-folder";
```

Comment:
You can make these constants to remove some duplication.

Reply:
I was thinking about that here and in the aspect filter test class. Will do.